src/services/item/plugins/geolocation/repository.ts

```diff
@@ -59,28 +62,54 @@ export class ItemGeolocationRepository {
       lng2,
       keywords,
     }: {
-      lat1: ItemGeolocation['lat'];
-      lat2: ItemGeolocation['lat'];
-      lng1: ItemGeolocation['lng'];
-      lng2: ItemGeolocation['lng'];
+      lat1?: ItemGeolocation['lat'];
+      lat2?: ItemGeolocation['lat'];
+      lng1?: ItemGeolocation['lng'];
+      lng2?: ItemGeolocation['lng'];
       keywords?: string[];
     },
+    parentItem?: Item,
   ): Promise<ItemGeolocation[]> {
-    const [minLat, maxLat] = [lat1, lat2].sort((a, b) => a - b);
-    const [minLng, maxLng] = [lng1, lng2].sort((a, b) => a - b);
+    // should include at least parentItem or all lat/lng
```

Comment:
It is possible to reflect this constraint in the type of the function to enforce to call it correctly ?
If it is too much code or too difficult, I think it is interesting to add a comment in the method's doc (Javadoc) to indicate this constraint.

Here could be a solution ? Maybe not !

```
type Boundaries = {
  lat1: number;
  lat2: number;
  lng1: number;
  lng2: number;
}

type ParentItem = string;

type GetItemsInParams =
  | { parentItem: ParentItem }
  | { boundaries: Boundaries }
  | { parentItem: ParentItem; boundaries: Boundaries };

const validateParams = (params: GetItemsInParams) => {
  if (!('parentItem' in params) && !('boundaries' in params)) {
    throw new Error('Either parentItem or boundaries is required');
  }
}
```